### DISCUSSION TITLE
Error in GENIE3

### DISCUSSION THREAD

**Discussion by KabitaBaral1** [aertslab/SCENIC#264] [R]
Hi,

I am running SCENIC on my dataset and the dataset is a big with dimension of 24110 X 28734

When I run my code, it gets stuck at GENIE3 part with the following error message:

Using 1092 TFs as potential regulators...
Running GENIE3 part 1
Error in weightMatrix[regulatorNames, ] <- weightMatrix.reg[regulatorNames,  : 
  number of items to replace is not a multiple of replacement length
Calls: runGenie3 -> <Anonymous> -> <Anonymous> -> .GENIE3
In addition: Warning message:
In mclapply(argsList, FUN, mc.preschedule = preschedule, mc.set.seed = set.seed,
  :
  scheduled cores 3, 5, 8, 9, 12, 13, 14, 15, 17, 20, 21, 23, 24, 25, 30, 33, 34
, 37, 39, 41, 42, 43, 44, 46, 48, 49, 52, 54 did not deliver results, all values
 of the jobs will be affected
Execution halted

I did check the previous issue posted with similar error and tried troubleshooting using different number of cores but that didn't work.

Could you please help me with this?

Here is my code snippet :

setwd("/home/kbaral/SCENIC/")

library("Seurat")
library("RcisTarget")
library("AUCell")
library("GENIE3")
library("SCENIC")

schwann <- readRDS("skin_nerve.rds")
cellInfo <- data.frame(CellType = schwann@active.ident, SC = schwann@meta.data$c
lusters, row.names = colnames(schwann))
dir.create("int")
saveRDS(cellInfo, file="int/cellInfo.Rds")

colVars <- list(CellType=setNames(c("lightcoral","darkgoldenrod",  "springgreen3
", "steelblue2","magenta2"), 
                                  c('skin0','nerve1','skin2','nerve3', "nerve4")
),
                SC= setNames(c("#04C0CC", "#EC7B70"), 
                             c("skin", "nerve")))
saveRDS(colVars, file="int/colVars.Rds")

schwann <- schwann@assays$RNA@counts
schwann <- as.matrix(schwann)
exprMat <- schwann
dim(exprMat)

org= "hgnc"
dbDir= "/home/kbaral/SCENIC/cisTarget_databases/"

scenicOptions <- initializeScenic(org= org, dbDir=dbDir, nCores=54)
scenicOptions@inputDatasetInfo$datasetTitle <- "schwann_cells"
scenicOptions@inputDatasetInfo$cellInfo <- "int/cellInfo.Rds"
scenicOptions@inputDatasetInfo$colVars <- "int/colVars.Rds"
saveRDS(scenicOptions, file="int/scenicOptions.Rds") 

genesKept <- geneFiltering(exprMat, scenicOptions=scenicOptions,
                           minCountsPerGene=3*.01*ncol(exprMat),
                           minSamples=ncol(exprMat)*.01)

exprMat_filtered <- exprMat[genesKept, ]
logMat <- log2(exprMat+1)
saveRDS(scenicOptions, file="int/scenicOptions.Rds")

exprMat_filtered <- log2(exprMat_filtered+1) 
dim(exprMat_filtered)
options(error = recover)
set.seed(123)

runCorrelation(exprMat_filtered, scenicOptions)
runGenie3(exprMat_filtered, scenicOptions)
runSCENIC_1_coexNetwork2modules(scenicOptions)
runSCENIC_2_createRegulons(scenicOptions)
runSCENIC_3_scoreCells(scenicOptions, logMat)

Thank you

**Comment by vbouther** [comment]
Hello, 
How much memory have you allowed to the job? 
I had the same issue and resolved it increasing allowed memory.
Hope this will work for you too!
